### PR TITLE
fix #6654 ensure non-submit buttons have type="button" 

### LINKED ISSUE #6654: enum buttons should be type="button"
The same problem as #6633 - if there are enum buttons on the page, and you select the Enter key expecting the Save Changes button to activate, the enum is swapping instead.

For example, changing the Distance field on this form and selecting the enter-key, causes the Sampling Method value to toggle to Automatic, rather than submitting the form.

x

### LINKED ISSUE #6656: sdds_beam sample_interval has wrong type
sdds_beam.sample_interval is currently marked as boolean. It should allow integer values to be entered.

image

### PR DESCRIPTION
**Pull request by moellep**
fix #6656 sdds_beam.sample_interval is non-boolean
Added sirepo.instrument_srw utility to debug SRW calls for CalcElecFieldSR, PropagElecField and CalcPartTraj.
Fixed layout issues with report notes display, changed to a srTooltip.
Removed unused reportCfg and plotSVG items.

### CODE REVIEW

**Comment by moellep** [comment]
@robnagler I also included the sirepo.instrument_srw utility in this PR - it is very useful for debugging SRW calls. Do you think it should go somewhere other than sirepo.instrument_srw? Maybe sirepo.debug.instrument_srw?

**Comment by robnagler** [comment]
> I also included the sirepo.instrument_srw utility in this PR - it is very useful for debugging SRW calls. Do you think it should go somewhere other than sirepo.instrument_srw? Maybe sirepo.debug.instrument_srw?

template.srw_debug seems more logical, since we start all template related modules with the code name. We keep everything in the template directories that are code related. I don't think debug needs a new package.

File: sirepo/package_data/static/js/sirepo-components.js [hunk 1]
```diff
@@ -2398,8 +2396,8 @@ SIREPO.app.directive('reportPanel', function(appState, utilities) {
         template: `
             <div class="panel panel-info" data-ng-attr-id="{{ ::reportId }}">
               <div class="panel-heading clearfix" data-panel-heading="{{ reportTitle() }}" data-model-key="modelKey" data-is-report="1" data-report-id="reportId"></div>
-              <div data-report-content="{{ reportPanel }}" data-model-key="{{ modelKey }}" data-report-id="reportId" data-report-cfg="reportCfg"><div data-ng-transclude=""></div></div>
-              <button data-ng-if="notes()" class="close sr-help-icon notes" title="{{ notes() }}"><span class="glyphicon glyphicon-question-sign"></span></button>
+              <div data-report-content="{{ reportPanel }}" data-model-key="{{ modelKey }}" data-report-id="reportId"><div data-ng-transclude=""></div></div>
+              <div data-ng-if="notes()"><span class="pull-right sr-notes" data-sr-tooltip="{{ notes() }}" data-placement="top"></span><div class="clearfix"></div></div>
```
Comment (mkeilman) [inline sirepo/package_data/static/js/sirepo-components.js:2400]: The icon is hard to spot on e.g. a shadow beam statistics report (it's in the same place in master and I have the same comment). Would top-left work, or as part of the panel title?

File: sirepo/package_data/static/js/sirepo-plotting.js [hunk 2]
```diff
@@ -1189,12 +1189,12 @@ SIREPO.app.directive('colorPicker', function(appState, panelState) {
...
+                <button type="button" class="dropdown-toggle sr-color-button" data-ng-style="bgColorStyle()" data-toggle="dropdown"></button>
                 <ul class="dropdown-menu">
                     <div class="container col-sm-8">
                         <div data-ng-repeat="r in range(rows) track by $index" class="row">
                             <li data-ng-repeat="c in range(cols) track by $index" style="display: inline-block">
-                                <button data-ng-if="pcIndex(r, c) < pickerColors.length" class="sr-color-button" data-ng-class="{\'selected\': getColor(model[field]).toUpperCase() == getPickerColor(r, c).toUpperCase()}" data-ng-style="bgColorStyle(getPickerColor(r, c))" data-ng-click="setColor(getPickerColor(r, c))"></button>
+                                <button type="button" data-ng-if="pcIndex(r, c) < pickerColors.length" class="sr-color-button" data-ng-class="{\'selected\': getColor(model[field]).toUpperCase() == getPickerColor(r, c).toUpperCase()}" data-ng-style="bgColorStyle(getPickerColor(r, c))" data-ng-click="setColor(getPickerColor(r, c))"></button>
```
Comment (mkeilman) [inline sirepo/package_data/static/js/sirepo-plotting.js:1197]: Not for this PR, but we can ditch this limited color picker in favor of the new one

File: sirepo/package_data/static/js/sirepo-components.js [hunk 3]
```diff
@@ -812,7 +813,7 @@ SIREPO.app.directive('fieldEditor', function(appState, keypressService, panelSta
               ${SIREPO.appFieldEditors}
               <div data-ng-switch-default data-ng-class="fieldClass">
                 <div data-ng-if="wantEnumButtons" class="btn-group">
-                  <button class="btn sr-enum-button" data-ng-repeat="item in enum[info[1]]" data-ng-click="model[field] = item[0]" data-ng-class="{\'active btn-primary\': isSelectedValue(item[0]), \'btn-default\': ! isSelectedValue(item[0])}">{{ item[1] }}</button>
+                  <button type="button" class="btn sr-enum-button" data-ng-repeat="item in enum[info[1]]" data-ng-click="model[field] = item[0]" data-ng-class="{\'active btn-primary\': isSelectedValue(item[0]), \'btn-default\': ! isSelectedValue(item[0])}">{{ item[1] }}</button>
```
Comment (mkeilman) [inline sirepo/package_data/static/js/sirepo-components.js:816]: Remove unneeded single-quote escapes

File: sirepo/template/srw_debug.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        if len(args) > i:
+            r.args[n] = _dump_value(args[i])
+    pkdlog("\n{}", pykern.pkjson.dump_pretty(r))
+
+
+def _dump_value(value):
+    if (
+        isinstance(value, int)
+        or isinstance(value, float)
+        or isinstance(value, str)
```
Comment (mkeilman) [inline sirepo/template/srw_debug.py:60]: Or `isinstance(value, (float, int, str))`...

**Review by mkeilman — COMMENTED** [review]
A few comments, otherwise cool

**Comment by moellep** [comment]
Sorry, @mkeilman I missed seeing your comments before I merged this PR. I'll add the changes to another PR. I don't think the reportCfg is the right way to pass this info - when you get to the energy plot in cloudmc, check with me and I bet we can get the info passed along without adding this back.

**Pull request merged**